### PR TITLE
Allow for more expressive comparisons and remove unneeded aliases.

### PR DESCRIPTION
**Pull request by dereuromark**
Resolves https://github.com/cakephp/chronos/issues/33

### CODE REVIEW

File: src/ChronosInterface.php [hunk 1]
```diff
@@ -336,7 +328,7 @@ public function min(ChronosInterface $dt = null);
      * @param ChronosInterface|null $dt The instance to compare with.
      * @return static
      */
-    public function max(ChronosInterface $dt = null);
+    public function maximum(ChronosInterface $dt = null);
```
Comment (markstory) [inline src/ChronosInterface.php]: min/max have equivalent functions in PHP stdlib.

Comment (dereuromark) [inline src/ChronosInterface.php]: So you think we should keep those?

**Comment by lorenzo** [comment]
I'm not sold on this. One of the goals of Chronos was being API compatible with Carbon so it could be replaced. Also, some of the names like `lte` and `gt` are already function names in the ORM. Personally I think people are used to those.

**Comment by dereuromark** [comment]
We can leave the old methods in as `@deprecated` for the transition to be smoother (and remove those in v2.0).
But I would rather have clear and descriptive methods than cryptic shortcuts that are not easy to read.

**Comment by ADmad** [comment]
I too am not sold on this. These short names are pretty common and most would know what they mean.

**Comment by markstory** [comment]
The short names also happen to be the same as those used in python for operator overloading (minus underscores).

**Comment by dereuromark** [comment]
I wonder how common that knowledge is in the PHP world :)
But all right, staying then it seems.

**Comment by burzum** [comment]
Can't we just have both? :)

**Comment by markstory** [comment]
@burzum I'm fine with more aliases as well.

**Comment by rchavik** [comment]
I'd rather not. Let's pick one and stick with it.

**Comment by ADmad** [comment]
I am not a fan of multiple names for the same thing either.

**Comment by dereuromark** [comment]
Hm, with https://github.com/briannesbitt/Carbon/pull/512/files being almost merged now we need to also consider adding them then (instead of replacing, we will have additional aliases).

IMO we should at least deprecate the ones we do not want to continue beyond 1.x.
This way we can easier clean up then in a future 2.x version.

**Comment by dereuromark** [comment]
This is merged now, btw: https://github.com/briannesbitt/Carbon/pull/512
Shall we then include them and mark them as `@deprecated` for the next major version?

**Comment by ADmad** [comment]
I am :-1: for marking shorter methods names as deprecated. Would rather live with aliasing.

**Comment by dereuromark** [comment]
I meant the longer ones (which have been added just now). We don't need to have all aliases in the future.

**Comment by markstory** [comment]
Having multiple aliases doesn't seem that bad. It lets people use the style they want which is nice. We can revisit whether or not we drop one set of aliases in the future I think.

**Comment by dereuromark** [comment]
OK, then we should add the aliases that Carbon introduced recently into this library for BC reasons, as well.

**Comment by ionas** [comment]
If chronos is meant to be a carbon replacement in future / out of cakephp BC scope. Yes.

Else my preferences are:
1. Long versions (deprecate shortcuts)
2. Short versions (deprecate long versions)
3. Having both

**Comment by dereuromark** [comment]
We probably now have to go with both, someone just needs to do the work and update it to a new PR.

**Comment by ionas** [comment]
I'd rather stick to either being fully carbon compatible or reduce the set of methods to one.
https://github.com/briannesbitt/Carbon/blob/master/src/Carbon/Carbon.php#L792

**Comment by dereuromark** [comment]
Deprecating sth is fully compatible still.

**Pull request closed without merging**